runtime/lua/vim/treesitter/languagetree.lua

```diff
@@ -363,6 +363,10 @@ function LanguageTree:_get_injections()
             table.insert(ranges, node)
           end
         end
+
+        if metadata[id] and metadata[id].range and #ranges == 0 then
+          table.insert(ranges, metadata[id].range)
+        end
```

Comment:
won't this still insert the original node into ranges?

Reply:
Yeah, I guess. Not fully sure the best place for this. What about line 337?

Reply:
you could create a function that resolved the node or the range `process_node(node, id, metadata)`. Then you could also `offset!` the `lang` by just appliying this function. nvim-treesitter's `get_node_text` also worked with nodes and `{range[1], range[2], range[3], range[4]}`, so that you could use everywhere a node or a range interchangeably.

Reply:
Sorry, but that mostly went over my head. I think the main problem is that I simply just don't understand this code. It looks pretty fragile and overly complex.

Reply:
Ok. I think I've done it.